### PR TITLE
Add a custom megazord for FFTV

### LINKED ISSUE #1560: Make a custom megazord for FFTV
Ref https://github.com/mozilla/application-services/issues/1559

FFTV only needs the FxA component and only on a single architecture, they shouldn't need to pull in the whole megazord on four different architectures.

┆Issue is synchronized with this [Jira Story](https://jira.allizom.org/browse/SYNC-256)

### PR DESCRIPTION
**Pull request by rfk**
Fixes #1560; Draft for now while we understand exact requirements.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `cargo test --all` produces no test failures
  - `cargo clippy --all --all-targets --all-features` runs without emitting any warnings
  - `cargo fmt` does not produce any changes to the code
  - `./gradlew ktlint detekt` runs without emitting any warnings
  - `swiftformat --swiftversion 4 megazords components/*/ios && swiftlint` runs without emitting any warnings or producing changes
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

**Comment by rfk** [comment]
@thomcc does this miss anything obvious w.r.t. adding a new megazord to the build?

File: megazords/fftv/android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    apiLevel = 21
+
+    // Where Cargo writes its outputs.
+    targetDirectory = '../../../target'
+
+    libname = 'fftv'
+
+    // The Cargo targets to invoke.  The mapping from short name to target
+    // triple is defined by the `rust-android-gradle` plugin.
+    targets = ['arm64']
```
Comment (thomcc) [inline megazords/fftv/android/build.gradle]: I guess they only need arm64?

File: megazords/fftv/android/build.gradle [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    // Perform release builds (which should have debug info, due to
+    // `debug = true` in Cargo.toml).
+    profile = "release"
+
+    extraCargoBuildArguments = rootProject.ext.extraCargoBuildArguments
+
+    exec = rootProject.ext.cargoExecWithSQLCipher
```
Comment (thomcc) [inline megazords/fftv/android/build.gradle]: They don't need SQLcipher, I believe.
```suggestion
    exec = rootProject.ext.cargoExec
```

File: megazords/fftv/android/build.gradle (hunk 1 above)
Comment (thomcc) [reply megazords/fftv/android/build.gradle]: Actually, even if they do, this probably still needs to be `rootProject.ext.rustTargets`.

Comment (rfk) [reply megazords/fftv/android/build.gradle]: Per https://github.com/mozilla/application-services/issues/1559 they only need a single platform, but I hadn't considered the `forUnitTest` variants...

Comment (thomcc) [reply megazords/fftv/android/build.gradle]: Not including x86 will also break running things under a simulator. If they're including the .so files for other architectures in the apk for a given arch, it's going to be easier to fix that on their end than adding special cases in our repo.

**Comment by rfk** [comment]
> This looks right. How much smaller is libfftv.so than libmegazord.so, just out of curiosity?

Mentioned on slack, but replying here for completeness: about 2M smaller so far.

**Comment by rfk** [comment]
Closing this, it's easy to bring back if we decide to do it.

**Pull request closed without merging**